draft.md

```diff
@@ -639,13 +661,13 @@ template <class U, class C=default_copy<U>,
   If the arguments `c` and/or `d` are not supplied, then `C` and/or `D`
   respectively are default constructible types that are not pointer types.
 
-* _Expects_: `C` and `D` meet the `Cpp17CopyConstructible` 
+* _Preconditions_: `C` and `D` meet the `Cpp17CopyConstructible` 
   and `Cpp17Destructible` requirements.
   
   Move-initialization of objects of type `C` and `D` does not exit 
   via an exception.
 
-  If `p` is non-null then the expression `c(*p)` has type `U*`. 
+  If `p` is non-null `invoke_result_t<c, remove_pointer_t<p>>` has type `U*`. 
```

Comment:
Responding to Daniel Krügler's comments: 

> Expects: I wonder about this part of this Precondition: "If p is non-null then the expression c(*p) has type U*" In which way is this not a Constraints: element such as "is_same_v<decltype(c(*p)), U*>"? Are you talking about dynamic types here? If so, this should be made clearer.

I hope moving to `invoke_result_t` here for the specification is clearer but opinions/feedback would be appreciated.

Reply:
Code as written LGTM